keras/applications/imagenet_utils.py

```diff
@@ -1,12 +1,18 @@
 import json
 import warnings
+import numpy as np
 
 from ..utils.data_utils import get_file
 from .. import backend as K
 
 CLASS_INDEX = None
 CLASS_INDEX_PATH = 'https://s3.amazonaws.com/deep-learning-models/image-models/imagenet_class_index.json'
 
+_IMAGENET_MEAN_CF = None
+_IMAGENET_MEAN_CF_4D = None
+_IMAGENET_MEAN_CL = None
```

Comment:
I see now that the proposed naming was confusing, because we also have to handle the case where there is no batch dimension. Let's go with explicit names to clear this up:

`_IMAGENET_MEAN_HWC` (3D)
`_IMAGENET_MEAN_CHW` (3D)
`_IMAGENET_MEAN_NHWC` (4D, channels last)
`_IMAGENET_MEAN_NCHW` (4D, channels first)

I believe you are currently not properly handling the NHWC case; please fix.